src/libsync/logger.cpp

```diff
@@ -52,8 +53,15 @@ namespace OCC {
 
 Logger *Logger::instance()
 {
-    static Logger log;
-    return &log;
+    static auto *log = [] {
+        auto log = new Logger;
+        qAddPostRoutine([] {
```

Comment:
Can't that be avoided by having the `close()` command in the destructor of the Logger?

I always have to look up these Qt special routines, and also ... as you mention ... which qPostRoutine() is called first? Can you have multiple at all?

Reply:
You can have multiple, the thing is that the watcher also uses the logger.